libparsec/crates/client/src/workspace/store/prevent_sync_pattern.rs

```diff
@@ -0,0 +1,118 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+use std::sync::Arc;
+
+use libparsec_platform_storage::workspace::{UpdateManifestData, WorkspaceStorage};
+use libparsec_types::{
+    prelude::{anyhow, thiserror},
+    DataError, LocalDevice, LocalFolderManifest, LocalWorkspaceManifest, SecretKey,
+};
+
+#[derive(Debug, thiserror::Error)]
+pub enum ApplyPreventSyncPatternError {
+    #[error("Decryption error")]
+    DecryptionError,
+    #[error(transparent)]
+    Internal(anyhow::Error),
+}
+
+pub(super) async fn ensure_prevent_sync_pattern_applied_to_wksp(
+    storage: &mut WorkspaceStorage,
+    device: Arc<LocalDevice>,
+) -> Result<(), ApplyPreventSyncPatternError> {
+    const PAGE_SIZE: u32 = 1000;
+
+    let (pattern, fully_applied) = storage
+        .get_prevent_sync_pattern()
+        .await
+        .map_err(ApplyPreventSyncPatternError::Internal)?;
+
+    if fully_applied {
+        return Ok(());
+    }
+
+    let mut offset = 0;
+    loop {
+        let manifests = storage
+            .list_manifests(offset, PAGE_SIZE)
+            .await
+            .map_err(ApplyPreventSyncPatternError::Internal)?;
+
+        let mut updated_manifest = Vec::with_capacity(manifests.len());
```

Comment:
if you want to optimize this code you should:
- don't set a capacity to vec
- move the vec creation outside of the loop (and add a clear at the end of the loop obviously ^^)

This way we avoid doing one allocation per loop iteration, and we most likely avoid having to do a 1000 elements allocation (considering file manifests are skipped, and among folders there is few with confinement, it is likely we only have a couple of manifests to modified for each batch of 1000 elements).

Reply:
done